bindata/network/kuryr/006-controller.yaml

```diff
@@ -0,0 +1,268 @@
+              else:
+                  subnets = neutron.list_subnets(name=subnet_name)
+              if not subnets['subnets']:
+                  body = {
+                      'name': subnet_name,
+                      'project_id': project_id,
+                      'ip_version': 4,
+                      'enable_dhcp': False,
+                      'network_id': network_id,
+                  }
+                  if subnetpool_id:
+                      body['subnetpool_id'] = subnetpool_id
+                  else:
+                      body['cidr'] = cidr
+
+                  if gateway_ip:
+                      body['gateway_ip'] = gateway_ip
+
+                  subnet_id = neutron.create_subnet(
+                      body={'subnet': body})['subnet']['id']
+              else:
+                  subnet_id = subnets['subnets'][0]['id']
+
+              return subnet_id
+
+
+          def create_sg(sg_name):
+              sgs = neutron.list_security_groups(name=sg_name)
+              if not sgs['security_groups']:
+                  sg_id = neutron.create_security_group(body={
+                      'security_group': {
+                          'name': sg_name,
+                          'project_id': project_id,
+                  }})['security_group']['id']
+              else:
+                  sg_id = sgs['security_groups'][0]['id']
+
+              return sg_id
+
+
+          def create_sg_rule(sg_id, rule):
+              try:
+                  body = {
+                      'project_id': project_id,
+                      'security_group_id': sg_id,
+                  }
+                  body.update(rule)
+                  neutron.create_security_group_rule(body={'security_group_rule': body})
+              except n_exc.Conflict:
+                  pass
+
+
+          data = {'neutron_defaults': {}}
+          svc_net_id = create_network('kuryr-service-network')
+          svc_net = netaddr.IPNetwork(svc_net_cidr)
+          svc_subnet_id = create_subnet('kuryr-service-subnet', svc_net_id,
+                                        cidr=svc_net_cidr, gateway_ip=svc_net[-2])
+          data['neutron_defaults']['service_subnet'] = svc_subnet_id
+
+          pod_net_id = create_network('kuryr-pod-network')
+          if mode == 'Multitenant':
+              pod_sp_id = create_subnetpool('kuryr-pod-subnetpool', pod_net_cidrs)
+              pod_subnet_id = create_subnet('kuryr-pod-subnet', pod_net_id,
+                                            subnetpool_id=pod_sp_id)
+              # FIXME(dulek): This is missing setting DNS server for the pod subnet.
+              data['neutron_defaults']['pod_subnet_pool'] = pod_sp_id
+          elif mode == 'Subnet':
+              pod_subnet_id = create_subnet('kuryr-pod-subnet', pod_net_id,
+                                            cidr=pod_net_cidrs[0])
+          data['neutron_defaults']['pod_subnet'] = pod_subnet_id
+
+          if mode == 'namespace':
```

Comment:
can mode be Multitenant and namespace? I believe the namespace mode also needs the pod_sg_id